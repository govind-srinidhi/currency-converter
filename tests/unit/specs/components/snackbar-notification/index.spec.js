/**
 * @file It contains unit tests for snackbar.
 */
 import { mount, createLocalVue } from "@vue/test-utils"
 import Vuetify from "vuetify"
 import SnackbarNotification from "@/components/snackbar-notification"

 let wrapper

 const snackBarNotificationComponent = () => wrapper.findComponent({ ref: "snackbar_notification" })
 const notifyIcon                    = () => wrapper.findComponent({ ref: "icon_notification" })
 const closeNotification             = () => wrapper.findComponent({ ref: "button_close" })
 
 const mountNotificationComponent = propsData => {
   const localVue = createLocalVue()

   localVue.use(Vuetify)
 
   const vuetify = new Vuetify()
 
   wrapper = mount(SnackbarNotification, {
     sync: false,
     localVue,
     propsData,
     vuetify
   })
 }

 
 describe("snackbar notification component", () => {
   beforeEach(() => {
     mountNotificationComponent({
       pNotification: {
        type: "error",
        icon: "mdi-alert-circle",
        text: "error notification"
       }
     })
   })
 
   afterEach(() => {
     jest.clearAllMocks()
   })
 
   it("renders the notification page", () => {
     expect(wrapper).toBeTruthy()
   })
 
   it("renders elements correctly", async () => {
     expect(snackBarNotificationComponent().exists()).toBeTruthy()
     expect(notifyIcon().exists()).toBeTruthy()
     expect(closeNotification().exists()).toBeTruthy()
   })

   it("close event to be called, when notification close icon is clicked", async () => {
     closeNotification().trigger("click")
     await wrapper.vm.$nextTick()
     expect(wrapper.emitted().close).toBeTruthy()
   })

   it("matches the default snapshot", async () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
 })