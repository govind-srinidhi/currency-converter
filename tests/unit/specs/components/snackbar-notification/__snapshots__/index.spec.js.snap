// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snackbar notification component matches the default snapshot 1`] = `
<div class="v-snack subtitle-2 v-snack--active v-snack--right v-snack--top" style="padding-bottom: 0px; padding-top: 0px;">
  <transition-stub name="v-snack-transition">
    <div class="v-snack__wrapper v-sheet v-sheet--outlined theme--light error--text">
      <div role="status" aria-live="polite" class="v-snack__content"><i aria-hidden="true" class="v-icon notranslate mr-2 mdi mdi-alert-circle theme--light error--text"></i>error notification</div>
      <div class="v-snack__action "><button type="button" class="v-btn v-btn--icon v-btn--round theme--light v-size--default error--text"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-close-circle theme--light"></i></span></button></div>
    </div>
  </transition-stub>
</div>
`;
